project.yaml

```diff
@@ -901,20 +901,62 @@ Layer:
       <<: *osm2pgsql
       table: |-
         (SELECT
-            way,
-            COALESCE(
-              ('highway_' || (CASE WHEN highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'cycleway', 'living_street', 'track', 'path', 'platform', 'services') THEN highway ELSE NULL END)),
-              ('railway_' || (CASE WHEN railway IN ('platform') THEN railway ELSE NULL END)),
-              (('aeroway_' || CASE WHEN aeroway IN ('runway', 'taxiway', 'helipad') THEN aeroway ELSE NULL END))
-            ) AS feature
-          FROM planet_osm_polygon
-          WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
-            OR railway IN ('platform')
-            OR aeroway IN ('runway', 'taxiway', 'helipad')
+            way, feature
+          FROM (
+            SELECT -- features mapped as polygons
+                way, 
+                COALESCE(
+                  ('highway_' || (CASE WHEN highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'cycleway', 'living_street', 'track', 'path', 'platform', 'services') THEN highway ELSE NULL END)),
+                  ('railway_' || (CASE WHEN railway IN ('platform') THEN railway ELSE NULL END)),
+                  ('aeroway_' || (CASE WHEN aeroway IN ('taxiway', 'helipad') THEN aeroway ELSE NULL END))
+                ) AS feature,
+                z_order, way_area FROM planet_osm_polygon
+              WHERE (highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
+                OR railway IN ('platform')
+                OR aeroway IN ('taxiway', 'helipad'))
+                AND way && !bbox!
+            UNION ALL
+            SELECT -- aeroway lines, converted to areas
+                way, feature, z_order, ST_Area(way) AS way_area
+              FROM
+                (SELECT 
+                    ST_Transform(
+                      ST_Buffer(
+                        ST_Transform( -- Use a projection that allows accurate buffering
+                          way,
+                          _ST_BestSRID(ST_Transform(way, 4326)::geography) -- Find the SRID to use
+                        ),
+                        0.5 * CASE -- Buffer by the half-width
+                          WHEN width ~ '^\d{1,3}(\.\d+)?$' -- Validate the width tag
+                            THEN LEAST(width::real, 150) -- Place a maximum on the width
```

Comment:
Is the maximum of 150 reasonable? I didn't have a look at the data to see what a good number would be, just that we didn't want to let through a 1km runway width tag.